chia/data_layer/data_layer.py

```diff
@@ -183,7 +184,7 @@ async def _validate_batch(
                 continue
             # Pick the latest root in our data store with the desired hash, before our already validated data.
             root: Optional[Root] = await self.data_store.get_last_tree_root_by_hash(
-                tree_id, record.root, max_generation
+                tree_id, None if record.root == bytes32([0] * 32) else record.root, max_generation
```

Comment:
I'm slightly afraid changing data layer representation can break more stuff. (i.e. functions in data layer thinking it's None, while it has changed to bytes32). I've just picked minimal changes to map the 2, but in the future I think it'll be good for them to match.

Reply:
also, this is not only in the creation case. You can get  bytes32([0] * 32 when someone deletes all the data, then commits on chain.

Reply:
There needs to be a definition of how an empty store is hashed since we have to put a hash on chain.  My impression is that @Quexington chose `0x000000...` for the on-chain representation and then that propagated out through the Python side of the wallet.  I think Bram mentioned the hash of nil being an option as well (I don't know what that ends up being).  I'm all for a consistent form in the Python code.  If the form is the on-chain hash then it should be defined in one place and that variable should be referenced (rather than writing `bytes32([0] * 32)` everywhere).

Reply:
i dont see why this should break stuff feels like it would help avoid issues by us not needing to swap out the nones in favor of the 0x000.. in all kinds of places

Reply:
we need to track in all data store code where we compare with None, and replace it with 0x000... it feels like a much bigger change than what I've done, so it has more opportunities to add extra bugs. although I agree in the long run, data store should use 0x0000.. as well